Documentation/Examples/RetriableSample.dib

```diff
@@ -0,0 +1,660 @@
+#### `Linear`
+This retries a specified number of times (specified by `maxTries`), with a constant delay between each retry (specified by `periodicity`).
+
+For example, `new Linear(TimeSpan.FromSeconds(1), 5)` will retry up to 5 times. Each time it will  delay by 1s. So the first retry will be after 1s (wall clock time 1s), the second after another 1s (wall clock time 2s), the third after another 1s (wall clock time 3s).
+
+#!csharp
+
+MockHttpService mockHttpService = new();
+Retriable.Retry(() => mockHttpService.MakeLotsOfRequests(), CancellationToken.None, strategy: new Linear(periodicity: TimeSpan.FromSeconds(1), maxTries: 5), policy: new RetryOnTooManyRequestsPolicy());
+
+#!markdown
+
+If we introduce a `Clock` object and add an overload for `MockHttpService.MakeLotsOfRequests()` that records the time elapsed for each retry since the initial execution, we can see the how the `Linear` retry policy retries over time.
+
+#!csharp
+
+public class Clock
+{
+    public int Count { get; private set; }
+    public Stopwatch stopwatch { get; }
+    
+    public Clock()
+    {
+        Count = 0;
+        stopwatch = new Stopwatch();
+    }
+
+    public void IncrementCounter()
+    {
+        this.Count += 1;
+    }
+}
+
+#!csharp
+
+public class MockHttpService
+{
+    public void MakeLotsOfRequests()
+    {
+        Console.WriteLine("MakeLotsOfRequests() method called");
+        throw new MockHttpServiceException("429 - Too Many Requests", "429");
+    }
+
+    public void MakeLotsOfRequests(Clock clock)
+    {
+        Console.WriteLine("MakeLotsOfRequests() method called");
+
+        if (clock.Count != 0)
+        {
+            clock.stopwatch.Stop();
+            TimeSpan timeElapsed = clock.stopwatch.Elapsed;
+            Console.WriteLine($"Time elapsed: {timeElapsed.Seconds}");
+        }
+
+        clock.IncrementCounter();
+        clock.stopwatch.Start();
+        throw new MockHttpServiceException("429 - Too Many Requests", "429");
+    }
+
+    public void MakeRequestThatWillTimeOut()
+    {
+        Console.WriteLine("MakeRequestThatWillTimeOut() method called");
+        throw new MockHttpServiceException("408 - Request timed out", "408");
+    }
+}
+
+#!csharp
+
+MockHttpService mockHttpService = new();
+Clock clock = new();
+Retriable.Retry(() => mockHttpService.MakeLotsOfRequests(clock), CancellationToken.None, strategy: new Linear(periodicity: TimeSpan.FromSeconds(1), maxTries: 5), policy: new RetryOnTooManyRequestsPolicy());
+
+#!markdown
+
+#### `Incremental`
+This retries a specified number of times, with an arithmetically increasing delay between each retry.
+
+For example, `new(maxTries: 5, intialDelay: TimeSpan.FromSeconds(1), step: TimeSpan.FromSeconds(1))` will retry up to 5 times. The initial delay will be one second. Each time it will increase the delay by 1s. So the first retry will be after 1s (wall clock time 1s), the second after another 2s (wall clock time 3s), the third after another 3s (wall clock time 6s).
```

Comment:
Is this another 'try' v. retry case?

Btw if they are *not* all consistent we should consider a breaking change to make them all consistent.